linkerd/app/core/src/errors.rs

```diff
@@ -193,3 +214,55 @@ impl std::fmt::Display for IdentityRequired {
 }
 
 impl std::error::Error for IdentityRequired {}
+
+impl metrics::LabelError<Error> for LabelError {
+    type Labels = Label;
+
+    fn label_error(&self, err: &Error) -> Self::Labels {
+        let reason = if err.is::<router::NoCapacity>() {
+            Reason::CacheFull
+        } else if err.is::<shed::Overloaded>() {
+            Reason::LoadShed
+        } else if err.is::<buffer::Aborted>() {
+            Reason::DispatchTimeout
+        } else if err.is::<IdentityRequired>() {
+            Reason::IdentityRequired
+        } else {
+            Reason::Unexpected
+        };
+
+        (self.0, reason)
+    }
+}
+
+impl metrics::FmtLabels for Reason {
+    fn fmt_labels(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(
+            f,
+            "message=\"{}\"",
+            match self {
+                Reason::CacheFull => "router full",
```

Comment:
```suggested
Reason::CacheFull => "router cache full",
```

I think it may be helpful to clarify that `router full` reasons happen *when* the router cache is full--especially for someone that may not be familiar with the proxy's router implementation and caching.

Reply:
In the upcoming branch that this change was derived from, caching is explicitly separated from routing. So I'll probably leave as-is on the assumption that lands soon.